sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -221,8 +221,7 @@ object ScalaNativePluginInternal {
             val isCppSource = path.endsWith(".cpp")
 
             val compiler = abs(if (isCppSource) clangpp else clang)
-            val flags = Seq("-O2") ++ (if (isCppSource) Seq("-std=c++11")
-                                       else Seq()) ++ opts
+            val flags    = (if (isCppSource) Seq("-std=c++11") else Seq()) ++ opts
```

Comment:
OK, I've updated the fix to fix the order so setting `nativeMode` to debug doesn't override it.